cpp/collections_module/algorithms/collections.hpp

```diff
@@ -0,0 +1,45 @@
+#include <mgp.hpp>
+#include <sstream>
+#include <string>
+
+constexpr std::string_view kReturnValueMin = "min";
+constexpr std::string_view kProcedureMin = "min";
+constexpr std::string_view kArgumentListMin = "list";
+
+void min(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
+  mgp::memory = memory;
+  const auto arguments = mgp::List(args);
+  const auto record_factory = mgp::RecordFactory(result);
+  try {
+    const mgp::List &list = arguments[0].ValueList();
+    if (list.Empty()) {
+      throw mgp::ValueException("Empty input list");
+    }
+    const mgp::Type &type = list[0].Type();
+    auto record = record_factory.NewRecord();
+
+    if (type == mgp::Type::Map || type == mgp::Type::Path || type == mgp::Type::List) {
+      std::ostringstream oss;
+      oss << type;
+      std::string s = oss.str();
+      throw mgp::ValueException("Unsuppported type for this operation, receieved type: " + s);
+    }
+
+    mgp::Value min = list[0];
+    for (size_t i = 0; i < list.Size(); i++) {
+      if (list[i].Type() != type && !(list[0].IsNumeric() && list[i].IsNumeric())) {
```

Comment:
what are you doing with list[0]? I think you can use move above and cache in some variable, `bool isListNumeric = list[0].isNumeric()`.

But do that before you move value from list